extensions/wikia/AdEngine/js/template/bfaa.js

```diff
@@ -1,48 +1,89 @@
 /*global define*/
 define('ext.wikia.adEngine.template.bfaa', [
+	'ext.wikia.adEngine.adContext',
 	'ext.wikia.adEngine.adHelper',
 	'wikia.document',
 	'wikia.log',
 	'wikia.window'
-], function (adHelper, doc, log, win) {
+], function (adContext, adHelper, doc, log, win) {
 	'use strict';
 
 	var logGroup = 'ext.wikia.adEngine.template.bfaa',
-		// SCSS property: $breakpoint-width-not-supported
-		breakPointWidthNotSupported = 767,
-		nav = doc.getElementById('globalNavigation'),
-		page = doc.getElementsByClassName('WikiaSiteWrapper')[0],
-		wrapper = doc.getElementById('WikiaTopAds');
-
-	function updateNavBar(height) {
-		var position = win.pageYOffset;
-
-		if (doc.body.offsetWidth <= breakPointWidthNotSupported || position <= height) {
-			wrapper.classList.add('bfaa-pinned-nav');
-			nav.classList.add('bfaa-pinned');
-		} else {
-			wrapper.classList.remove('bfaa-pinned-nav');
-			nav.classList.remove('bfaa-pinned');
+		breakPointWidthNotSupported = 767, // SCSS property: $breakpoint-width-not-supported
+		desktopHandler,
+		mobileHandler,
+		nav,
+		page,
+		wrapper;
+
+	desktopHandler = {
+		updateNavBar: function (height) {
+			var position = win.pageYOffset;
+
+			if (doc.body.offsetWidth <= breakPointWidthNotSupported || position <= height) {
+				wrapper.classList.add('bfaa-pinned-nav');
+				nav.classList.add('bfaa-pinned');
+			} else {
+				wrapper.classList.remove('bfaa-pinned-nav');
+				nav.classList.remove('bfaa-pinned');
+			}
+		},
+
+		show: function (height, backgroundColor) {
+			nav.style.top = '';
+			page.classList.add('bfaa-template');
+			wrapper.style.background = backgroundColor;
+
+			this.updateNavBar(height);
+			doc.addEventListener('scroll', adHelper.throttle(function () {
+				this.updateNavBar(height);
+			}.bind(this), 100));
+
+			if (win.WikiaBar) {
+				win.WikiaBar.hideContainer();
+			}
 		}
-	}
+	};
 
-	function show(params) {
-		var backgroundColor = params.backgroundColor ? '#' + params.backgroundColor.replace('#', '') : '#000',
-			height = params.height || 0;
+	mobileHandler =  {
+		show: function (height, backgroundColor) {
+			win.Mercury.Modules.Ads.getInstance().setSiteHeadOffset(height);
+			page.style.paddingTop = height + 'px';
+			wrapper.style.background = backgroundColor;
+			page.classList.add('bfaa-template');
 
-		nav.style.top = '';
-		page.classList.add('bfaa-template');
-		wrapper.style.background = backgroundColor;
+			adContext.addCallback(function () {
+				page.classList.remove('bfaa-template');
+				wrapper.style.background = '';
+				page.style.paddingTop = '';
+				win.Mercury.Modules.Ads.getInstance().setSiteHeadOffset(0);
```

Comment:
`getInstance()` is defined here:
https://github.com/Wikia/mercury/blob/dev/front/common/modules/ads.js#L54

`Ads.instance` is defined here:
https://github.com/Wikia/mercury/blob/dev/front/common/modules/ads.js#L309

"Static properties" in ES6 are described here:
http://exploringjs.com/es6/ch_classes.html#_static-data-properties

So I would say that yes, instance returned here is always the same object :) Moving it to a variable should work without any issues.